keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -253,14 +274,28 @@ nitty-gritty.
 -->
 
 ### Overview
-Each cluster in a ClusterSet will be assigned a unique identifier, that lives at least as long as that cluster is a member of the given ClusterSet, and is immutable for that same lifetime. This identifier will be stored in a new cluster-scoped `ClusterProperty` CR with the well known name `cluster.clusterset.k8s.io` that may be referenced by workloads within the cluster. The identifier must either:
+Each cluster in a ClusterSet will be assigned a unique identifier, that lives at
+least as long as that cluster is a member of the given ClusterSet, and is
+immutable for that same lifetime. This identifier will be stored in a new
+cluster-scoped `ClusterProperty` CR with the well known name
+`cluster.clusterset.k8s.io` that may be referenced by workloads within the
+cluster. The identifier must either:
 - be a valid [RFC-1123](https://tools.ietf.org/html/rfc1123) DNS label,
-- or be composed of two valid [RFC-1123](https://tools.ietf.org/html/rfc1123) DNS labels separated with a dot.
-The identifier may be created by an implementation dependent mechanism.
-
-While a member of a ClusterSet, a cluster will also have an additional `clusterset.k8s.io ClusterProperty` which describes its current membership. This property must be present exactly as long as the cluster's membership in a ClusterSet lasts, and removed when the cluster is no longer a member.
-
-More detail and examples of the uniqueness, lifespan, immutability, and content requirements for both the `cluster.clusterset.k8s.io ClusterProperty` and `clusterset.k8s.io ClusterProperty` are described further below. The goal of these requirements are to provide to the MCS API a cluster id of viable usefulness to address known user stories without being too restrictive or prescriptive.
+- or be composed of two valid [RFC-1123](https://tools.ietf.org/html/rfc1123)
```

Comment:
Contentious: Because we are no interacting with external control planes - is this sufficiently flexible?  If it turned out not to be, could we extend it?  DNS is more flexible than we give it credit for - underscores are not so uncommon as I once thought, for example!

Reply:
I propose we start with the current proposal, then add flexibility in beta if needed as it would not be a backward incompatible change.  It seems like the next logical step would be "any valid domain name" and I don't think we're ready or see sufficient need to impose that yet

Reply:
How about we change it from "one or two labels" to somewthing like:

"""
The identifier must be a valid RFC-1123 DNS subdomain and should be less than 128 characters in total.  This identifier might be used to compose larger DNS names (e.g. in the case of multi-cluster services), so care should be take to ensure that the final names fit into the limit of 253 characters.
"""